pkg/sources/reconciler/azureservicebussource/reconciler.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2023 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azureservicebussource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/reconciler"
+
+	commonv1alpha1 "github.com/triggermesh/triggermesh/pkg/apis/common/v1alpha1"
+	"github.com/triggermesh/triggermesh/pkg/apis/sources/v1alpha1"
+	reconcilerv1alpha1 "github.com/triggermesh/triggermesh/pkg/client/generated/injection/reconciler/sources/v1alpha1/azureservicebussource"
+	listersv1alpha1 "github.com/triggermesh/triggermesh/pkg/client/generated/listers/sources/v1alpha1"
+	common "github.com/triggermesh/triggermesh/pkg/reconciler"
+	"github.com/triggermesh/triggermesh/pkg/reconciler/event"
+	"github.com/triggermesh/triggermesh/pkg/sources/auth"
+	"github.com/triggermesh/triggermesh/pkg/sources/client/azure/servicebus"
+)
+
+// Reconciler implements controller.Reconciler for the event source type.
+type Reconciler struct {
+	// Getter than can obtain clients for interacting with Azure APIs
+	cg servicebus.ClientGetter
+
+	// Event Hubs adapter
+	base       common.GenericDeploymentReconciler[*v1alpha1.AzureServiceBusSource, listersv1alpha1.AzureServiceBusSourceNamespaceLister]
+	adapterCfg *adapterConfig
+}
+
+// Check that our Reconciler implements Interface
+var _ reconcilerv1alpha1.Interface = (*Reconciler)(nil)
+
+// Check that our Reconciler implements Finalizer
+var _ reconcilerv1alpha1.Finalizer = (*Reconciler)(nil)
+
+// ReconcileKind implements Interface.ReconcileKind.
+func (r *Reconciler) ReconcileKind(ctx context.Context, o *v1alpha1.AzureServiceBusSource) reconciler.Event {
+	// inject source into context for usage in reconciliation logic
+	ctx = commonv1alpha1.WithReconcilable(ctx, o)
+
+	// In the case when TopicID is present, it is necessary to ensure the subscription.
+	// when dealing with QueueID, there's no need.
+	if o.Spec.TopicID != nil {
```

Comment:
The queue case will leave the status set to `Unknown` ...

If that is the case we should either try to make the status conditions dynamic (not sure if that is something that we have done at any other component), or set the subscribed condition to true with a reason that says `NOTNEEDED`.

Reply:
It should be fixed now